std/special/bootstrap.zig

```diff
@@ -123,12 +123,15 @@ inline fn callMain() u8 {
         },
         builtin.TypeId.Int => {
             if (@typeOf(root.main).ReturnType.bit_count != 8) {
-                @compileError("expected return type of main to be 'u8', 'noreturn', 'void', or '!void'");
+                @compileError("expected return type of main to be 'u8', 'noreturn', 'void', '!void', or '!u8'");
             }
             return root.main();
         },
         builtin.TypeId.ErrorUnion => {
-            root.main() catch |err| {
+            const PayloadType = @typeOf(root.main).ReturnType.Payload;
+            if (@typeId(PayloadType) != builtin.TypeId.Void and @typeId(PayloadType) != builtin.TypeId.Int)
```

Comment:
This `Int` check is a bit lenient. I would check if it is exactly a u8 explicitly. This will be caught on the return of this function still, but you would not get the nicer compile error.

This actually occurs right now if you use an `i8` as the return value for the above branch.

Reply:
Thanks for spotting that